CosmicsCalib/src/CaloCosmicEnergy_module.cc

```diff
@@ -0,0 +1,746 @@
+
+
+    //we create arrays where we store informations of crystals that pass the selections
+ 
+    float PosX[nCrystals];
+    float PosY[nCrystals];
+    float ErrPos[nCrystals];
+    int IDs[nCrystals];
+    float path[nCrystals];
+    int whichHit[nCrystals];
+
+  };//end class
+
+  CaloCosmicEnergy::CaloCosmicEnergy(fhicl::ParameterSet const& pset) : 
+    art::EDAnalyzer(pset), 
+    _diagLevel(pset.get<int>("diagLevel", 0)),
+    CutNCryHit(pset.get<int>("CutNCryHit", 3)),
+    CutEnergyDep(pset.get<float>("CutEnergyDep", 10.)),
+    CutChi2Norm(pset.get<float>("CutChi2Norm", 2.5)),
+    _caloClusterTag(pset.get<art::InputTag>("caloClusterCollection")),
+    _caloHitTag(pset.get<art::InputTag>("caloHitCollection")){}
+
+  void CaloCosmicEnergy::beginJob(){
+    if (_diagLevel > 0 ) std::cout<< "CaloCosmicEnergy: Entering beginJob" << std::endl;
+ 
+    for(int iCry=0; iCry < nDisks*nCrystals; iCry++){
+      for(int iSiPM=0; iSiPM < nSiPMs; iSiPM++){
+	hSiPM[iCry][iSiPM] = tfdir.make<TH1F>(Form("Crystal_%i_SiPM_%i",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d",iSiPM, iCry),70, 15., 50.);
+	hSiPMv[iCry][iSiPM] = tfdirv.make<TH1F>(Form("Crystal_%i_SiPM_%i_v",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d only vertical tracks",iSiPM, iCry),70, 15., 50.);
+	hSiPMd[iCry][iSiPM] = tfdird.make<TH1F>(Form("Crystal_%i_SiPM_%i_d",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d only diagonal tracks",iSiPM, iCry),70, 15., 50.);
+	hSiPMfp[iCry][iSiPM] = tfdirfp.make<TH1F>(Form("Crystal_%i_SiPM_%i_fp",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d normalized track length",iSiPM, iCry),70, 15., 50.);
+      }
+      for(int ibin=0; ibin < Ebin; ibin++){
+	CryALR[iCry][ibin] = new TH1F(Form("Asymmetry_Crystal_%i_band_%i", iCry, ibin), Form("Asymmetry of crystal %i in band [%i, %i)",iCry, (ibin+1)*5, (ibin+1)*5+5), 200,-1., 1.);
+      }
+    }
+    MPV = tfdir_res.make<TH1F>("MPV_distr","MPV distribution",400, 10., 30.);
+    MPVv = tfdir_res.make<TH1F>("MPV_v_distr","MPV distribution only vertical tracks",400, 10., 30.);
+    MPVd = tfdir_res.make<TH1F>("MPV_d_distr","MPV distribution only diagonal tracks",400, 10., 30.);
+    MPVfp = tfdir_res.make<TH1F>("MPV_fp_distr","MPV distribution all normalized tracks",400, 10., 30.);
+    ALR_all = tfdir_alr.make<TH2F>("ALR","Asymmetry left right all energies",600,0.,600., 100, -1., 1.);
+    LR_all = tfdir_alr.make<TH1F>("ALR_All_Energies", "Asymmetry Left Right all energies", 200, -1, 1.);
+    for(int i =0; i<Ebin; i++){
+      LR[i] = tfdir_alr.make<TH1F>(Form("LR_ene_band_%i",i), Form("Ratio Left Right Energy band [%i - %i) MeV", i*5, (i+1)*5), 100, 0.5, 1.5);
+      ALR[i] = tfdir_alr.make<TH1F>(Form("ALR_ene_band_%i",i), Form("Asymmetry Left-Right Energy band [%i - %i) MeV", i*5, (i+1)*5), 200, -1., 1.);
+    }
+  }//end begin job
+
+
+  void CaloCosmicEnergy::analyze(art::Event const& event){
+    art::ServiceHandle<GeometryService> geom;
```

Comment:
I suggest omitting this entire block.  The GeomHandle in the next line will throw an exception with a useful error message if it cannot find the calorimeter.